lib/improver/cli/apply_emos_coefficients.py

```diff
@@ -130,58 +130,104 @@ def main(argv=None):
 
     args = parser.parse_args(args=argv)
 
+    # Load Cubes
     current_forecast = load_cube(args.forecast_filepath)
     coeffs = load_cube(args.coefficients_filepath)
 
-    original_current_forecast = current_forecast.copy()
+    # Process Cube
+    result = process(current_forecast, coeffs, args)
+
+    # Save Cube
+    save_netcdf(result, args.output_filepath)
+
 
+def process(current_forecast, coeffs, num_realizations=None,
+            random_ordering=False, random_seed=None,
+            ecc_bounds_warning=False, predictor_of_mean='mean'):
```

Comment:
Yes, I think this is exactly the pattern we should follow - cubes/files-like (and possibly coordinate-like) inputs as positional/required arguments and other options as keyword arguments. (Alas, we should do that for the plugins too - just use functions if there's no genuine state to be stored on the object.)